language/extensions/move-table-extension/src/lib.rs

```diff
@@ -381,25 +385,29 @@ fn native_new_table_handle(
     let mut table_data = table_context.table_data.borrow_mut();
 
     // Take the transaction hash provided by the environment, combine it with the # of tables
-    // produced so far, sha256 this and select 16 bytes from the result. Given the txn hash
+    // produced so far, sha256 this to produce a unique handle. Given the txn hash
     // is unique, this should create a unique and deterministic global id.
```

Comment:
One thing that might be worth thinking about: The assumption that TXN hashes are unique might **not** always hold. For example, in Bitcoin, different coinbase TXNs for different blocks could be identical and therefore have the same hash.